dhis-2/dhis-api/src/main/java/org/hisp/dhis/util/ObjectUtils.java

```diff
@@ -193,4 +195,34 @@ public static <T, U extends RuntimeException> T throwIfNull( T object, Supplier<
 
         return object;
     }
+
+    /**
+     * Util method that always returns a new Set, either instantiated from a
+     * non-null Set passed as an argument, or if a null arg is passed then
+     * returning an empty Set. This helps reduce possible NullPointerExceptions
+     * when trying to instantiate a Set with a null value.
+     *
+     * @param set
+     * @return
+     * @param <T>
+     */
+    public static <T> Set<T> newSetFromObjectOrEmpty( Set<T> set )
```

Comment:
There is a `List.copyOf` and `Set.copyOf` in the JDK. Is there a problem with immutable collections? If we want a mutable version of these I think we should also name them `copyOf`. It would also be good for such utility methods if you add the `@CheckForNull` and `@Nonnull` annotations to the signature especially if the method motivation is just about the null problem.

Reply:
`List/Set.copyOf` throws a `NPE` if any element is `null` or if the original collection is `null`. I can add those annotations and rename as well.